src/main/reactjs/src/components/publicTranslator/PublicTranslatorListing.tsx

```diff
@@ -1,44 +1,90 @@
-import { FC } from 'react';
 import {
   TableCell,
   Checkbox,
   TableHead,
   TableRow,
   Button,
 } from '@mui/material';
-import { TFunction } from 'i18next';
 import { Box } from '@mui/system';
 
 import { H2, H3, Text } from 'components/elements/Text';
 import { PaginatedTable } from 'components/tables/Table';
 import { ProgressIndicator } from 'components/elements/ProgressIndicator';
 import { PublicTranslator } from 'interfaces/translator';
-import { Selectable } from 'interfaces/selectable';
 import { APIResponseStatus } from 'enums/api';
 import { useAppDispatch, useAppSelector } from 'configs/redux';
 import { useAppTranslation } from 'configs/i18n';
 import {
+  addPublicTranslatorFilterError,
   addSelectedTranslator,
   removeSelectedTranslator,
 } from 'redux/actions/publicTranslator';
 import { publicTranslatorsSelector } from 'redux/selectors/publicTranslator';
-import { UIStates } from 'enums/app';
+import { UIStates, Severity, Filter } from 'enums/app';
 import { displayUIState } from 'redux/actions/navigation';
+import { showNotifierToast } from 'redux/actions/notifier';
+import { Utils } from 'utils/index';
 
-const getPublicTranslatorRow = (
+const getRowDetails = (
   translator: PublicTranslator,
-  t: TFunction,
-  selectionProps: Selectable
+  selected: boolean,
+  toggleSelected: () => void
 ) => {
+  return (
+    <ListingRow
+      translator={translator}
+      selected={selected}
+      toggleSelected={toggleSelected}
+    />
+  );
+};
+
+const ListingRow = ({
+  translator,
+  selected,
+  toggleSelected,
+}: {
+  translator: PublicTranslator;
+  selected: boolean;
+  toggleSelected: () => void;
+}) => {
+  // I18n
+  const { t } = useAppTranslation({
+    keyPrefix: 'akt.component.publicTranslatorFilters',
+  });
+
+  // Redux
+  const dispatch = useAppDispatch();
+  const { filters } = useAppSelector(publicTranslatorsSelector);
+  const { fromLang, toLang } = filters;
   const { firstName, lastName, languagePairs, town, country } = translator;
-  const { selected, toggleSelected } = selectionProps;
   const townInfo = `${town}${country ? `, ${country}` : ''}`;
 
+  const handleRowClick = () => {
+    const langFields = [Filter.FromLang, Filter.ToLang];
+
+    // Dispatch an error if the langpairs are not defined
+    langFields.forEach((field) => {
+      if (!filters[field] && !filters.errors?.includes(field))
+        dispatch(addPublicTranslatorFilterError(field));
+    });
+
+    if (filters?.errors?.length || !fromLang || !toLang) {
```

Comment:
Pitäisikö `filters?.errors?.length` olla oma käsittelynsä? Tai vaihtoehtoisesti sen voisi ehkä ottaa tässä vaiheessa pois? Jos tulevaisuudessa todetaankin että jokin muu tilanne kuin fromLang tai toLang puuttuminen on virhetilanne, tätä koodia pitää muuttaa - muuten laukeaa virhe hämäävällä virheviestillä.

Reply:
Hmm en ihan ymmärtänyt, miksi sillä olisi oma käsittely?

Reply:
Sitä ei pysty ottamaan pois muistaakseni, koska jos on toLang tai fromLang erroreja ei saa jatkaa eteenpäin.

Reply:
Tarkoitin jotain tällaista
```
if (!fromLang || !toLang) { ... } 
else if (filters?.errors?.length) 
  { dispatch(showGenericError); 
} else 
{ // ok }```

Reply:
Mutta, mikä tässä olisi pointtina? Saman herjaa halutaan näyttää kuitenkin jos kieliparit eivät ole määriteltynä?

Reply:
Jos toki tulevaisuudessa tulee sellainen tilanne niin silloin voisi erottaa silloin :)

Reply:
1) Jos `!fromLang || !toLang` riittää kertomaan että virhe on olemassa, niin eikö `filters?.errors?.length` kannata jättää pois if-lauseesta.
2) Jos tulevaisuudessa onkin muita mahdollisia virhetilanteita kuin kieliparin puuttuminen, silloin tuo `filters?.errors?.length || ..` tarttuu kiinni tällaiseenkin virheeseen ja näytetään väärä virheviesti (`selectLanguagePair`).

Reply:
Niin en tie vaikuttaako tämä loppujen lopuksi johonkin, mutta tarkoituksena oli estää niitä toimintoja jo tässä vaiheessa vaikka niitä ei täysin tiedetä.

Reply:
Otan kuitenkin sen pois :)